widgetssdk/src/test/java/com/glia/widgets/callvisualizer/ActivityWatcherForDialogsTest.kt

```diff
@@ -93,4 +100,54 @@ internal class ActivityWatcherForDialogsTest {
         activityWatcherForDialogs.dialogCallback?.emitDialogState(state)
         assertNotNull(activityWatcherForDialogs.alertDialog)
     }
+
+    @Test
+    fun mediaProjectionObjects_null_whenChatActivity() {
+        val activity = mock(ChatActivity::class.java)
+        activityWatcherForDialogs.onActivityPreCreated(activity, null)
+
+        assertNull(activityWatcherForDialogs.mediaProjectionManager)
+        assertNull(activityWatcherForDialogs.startMediaProjection)
+    }
+
+    @Test
+    fun mediaProjectionObjects_null_whenCallActivity() {
+        val activity = mock(CallActivity::class.java)
+        activityWatcherForDialogs.onActivityPreCreated(activity, null)
+
+        assertNull(activityWatcherForDialogs.mediaProjectionManager)
+        assertNull(activityWatcherForDialogs.startMediaProjection)
+    }
+
+    @Test
+    fun mediaProjectionObjects_noCreation_whenNotComponentActivity() {
+        val activity = mock(Activity::class.java)
+        activityWatcherForDialogs.registerForMediaProjectionPermissionResult(activity)
+
+        verify(activity, never()).getSystemService(MediaProjectionManager::class.java)
+    }
+
+    @Test
+    fun mediaProjectionObjects_creation_whenComponentActivity() {
+        val activity = mock(ComponentActivity::class.java)
+        activityWatcherForDialogs.registerForMediaProjectionPermissionResult(activity)
+
+        verify(activity, times(1)).getSystemService(MediaProjectionManager::class.java)
```

Comment:
NIT you don't need the .times(1) as it is implicit. If it is called twice, then the test will fail and if it is not called, it will fail. times(1) is the default value